src/validation.cpp

```diff
@@ -3766,6 +3767,9 @@ bool Chainstate::InvalidateBlock(BlockValidationState& state, CBlockIndex* pinde
         }
 
         InvalidChainFound(to_mark_failed);
+        if (pindex_was_in_chain) {
+            failed_prev_verification_progress = m_chainman.GuessVerificationProgress(to_mark_failed->pprev);
+        }
```

Comment:
fa53098472521de9d5b3cc42983043c240b7c321

Why is this here and not in the next block that is already guarded by `pindex_was_in_chain`? `to_mark_failed` and `m_chainman` are still in scope.